addons/mail/static/tests/discuss_app/discuss.test.js

```diff
@@ -737,24 +742,24 @@ test("Unfollow message", async function () {
         contains: [[".o-mail-Message-header small", { text: "on Thread followed" }]],
     });
     await contains(".o-dropdown-item:contains('Unfollow')");
-    await contains(".o-mail-Message:eq(2) [title='Expand']", { count: 0 });
+    await click(".o-mail-Message:eq(2) [title='Expand']");
     await contains(".o-mail-Message:eq(2)", {
         contains: [[".o-mail-Message-header small", { text: "on Thread not followed" }]],
     });
-    await contains(".o-mail-Message:eq(2) [title='Unfollow']", { count: 0 });
+    await contains(".o-dropdown-item:contains('Unfollow')", { count: 0 });
     await click(".o-mail-Message:eq(0) [title='Expand']");
     await click(".o-dropdown-item:contains('Unfollow')");
     await contains(".o-mail-Message", { count: 2 }); // Unfollowing message 0 marks it as read -> Message removed
     await contains(".o-mail-Message:eq(0)", {
         contains: [[".o-mail-Message-header small", { text: "on Thread followed" }]],
     });
-    await contains(".o-mail-Message:eq(0) [title='Expand']", { count: 0 });
-    await contains(".o-mail-Message:eq(0) [title='Unfollow']", { count: 0 });
+    await click(".o-mail-Message:eq(0) [title='Expand']");
+    await contains(".o-dropdown-item:contains('Unfollow')", { count: 0 });
```

Comment:
Click on "Expand" and immediately assert no "Unfollow" is not good, because this is prone to potential non-deterministic failure:

Let's say the "Unfollow" is actually there, if we click on "Expand", the await doesn't mean the dropdown is visible right away. It can time a few ms. During this time, `count: 0` can pass because the dropdown is not open at all.

Negative assertions are quite challenging to make them work with a good threshold of reliability. Here to be in a good enough state, we should await a positive `contains` which is very likely to wait enough time to assert the dropdown. One way is to await another `dropdown-item` like the `Reply`.

Risk is not 0 but it's very unlikely to render "Reply" dropdown without the "Unfollow" one, so that's good enough ;)